### ISSUE TITLE
"Technic icon NOT found" Even with icon.png inside the itlt config folder

### ISSUE THREAD

**Issue by martinmaroyan** [zlepper/itlt#19]
I set useTechnicDisplayName to true, added icon.png to the itlt config folder, but the game output still says 
[16:54:37] [Client thread/INFO] [itlt]: Technic icon NOT found.

MC Version: 1.12.2
OS Name: MacOS
Installed mods: replaymod, ido, itlt, surge, wynntils, phosphor-lighting
Icon: ![icon](https://user-images.githubusercontent.com/48408674/113491297-b311da00-949d-11eb-9db8-ff4a20cceabe.png)
Config:
```
# Configuration file

bitdetection {
    # If ShouldYellAt32BitUsers is set to true, this is the message that will be displayed to the user.
    S:ErrorMessage=You are using a 32 bit version of java. This is not recommended with this modpack.

    # Set to true to make itlt yell at people attempting to use 32x java for the modpack.
    B:ShouldYellAt32BitUsers=true
}


display {
    # Set to true to make minecraft attempt to maximize itself on startup (This is kinda unstable right now, so don't trust it too much)
    B:ShouldMaximizeDisplay=false

    # Set to true to load a custom icon from config/itlt/icon.png
    B:loadCustomIcon=true

    # Set to true to attempt to get the display name of the pack of the info json file 
    # This will take priority over windowDisplayTitle
    B:useTechnicDisplayName=true

    # Set to true to attempt to use the icon assigned to the modpack by the technic launcher. 
    # This will take priority over loadCustomIcon
    B:useTechnicIcon=true

    # Change this value to change the name of the MineCraft window
    S:windowDisplayTitle=Minecraft 1.12.2-forge-14.23.5.2854
}


server {
    # Set to true to have a dedicated server added to the server list ingame. The server will not overwrite others servers.
    B:AddDedicatedServer=false

    # The ip of the dedicated server to add.
    S:ServerIP=127.0.0.1:25555

    # The name of the dedicated server to add.
    S:ServerName=Localhost
}
```

**Comment by PaintNinja** [comment]
If you leave loadCustomIcon on but turn off useTechnicIcon, does it work then?

I think the useTechnicIcon option overrides the custom provided one if enabled.

**Comment by martinmaroyan** [comment]
Yea so I changed it to 
```
# Configuration file

bitdetection {
    # If ShouldYellAt32BitUsers is set to true, this is the message that will be displayed to the user.
    S:ErrorMessage=You are using a 32 bit version of java. This is not recommended with this modpack.

    # Set to true to make itlt yell at people attempting to use 32x java for the modpack.
    B:ShouldYellAt32BitUsers=true
}


display {
    # Set to true to make minecraft attempt to maximize itself on startup (This is kinda unstable right now, so don't trust it too much)
    B:ShouldMaximizeDisplay=false

    # Set to true to load a custom icon from config/itlt/icon.png
    B:loadCustomIcon=true

    # Set to true to attempt to get the display name of the pack of the info json file 
    # This will take priority over windowDisplayTitle
    B:useTechnicDisplayName=true

    # Set to true to attempt to use the icon assigned to the modpack by the technic launcher. 
    # This will take priority over loadCustomIcon
    B:useTechnicIcon=false

    # Change this value to change the name of the MineCraft window
    S:windowDisplayTitle=Wynncraft
}


server {
    # Set to true to have a dedicated server added to the server list ingame. The server will not overwrite others servers.
    B:AddDedicatedServer=false

    # The ip of the dedicated server to add.
    S:ServerIP=127.0.0.1:25555

    # The name of the dedicated server to add.
    S:ServerName=Localhost
}
```
and now in the console it says `[15:15:08] [Client thread/INFO] [itlt]: Custom modpack icon found`
which is good, but after that there is an error `[15:15:08] [Client thread/INFO] [itlt]: Cache file not found.`
The icon still stays the same "crafting table". and also the window display title didn't change anything, the name is still `java`

**Comment by PaintNinja** [comment]
Okay, it seems like the Technic Launcher detection is failing which is the cause of the icon and window title not being set based on your modpack's details on the Technic Platform. Where is the launcher and your modpack installed?

You can disable useTechnicDisplayName and it should use the "Wynncraft" windowDisplayTitle you've set. As for the icon and window title being default and "java", I believe this is specific to using MacOS in combination with a much older version of Minecraft (1.12 is almost 4 years old at this point). What version of MacOS are you running? What version of Java are you using?

**Comment by martinmaroyan** [comment]
`Java is Java HotSpot(TM) 64-Bit Server VM, version 1.8.0_74, running on Mac OS X:x86_64:10.16`
MacOS Big Sur
The modpack is actually not installed in the main folder, its installed in a folder called "Wynncraft" inside `~/Application\ Support/minecraft/profiles`, and in the launcher i specified the game directory to be `~/Application\ Support/minecraft/profiles/Wynncraft`

**Comment by PaintNinja** [comment]
If I remember correctly, Java 8u74 from 2016 does not fully support modern MacOS, which may explain why the window title and icon are not applying for you.

You can get the latest update to Java 8 (currently 8u282) here which might help: https://adoptopenjdk.net/?variant=openjdk8&jvmVariant=hotspot if not, you may need to update Minecraft to a newer version (e.g. MC 1.14) that supports more modern versions of Java such as Java 11 or Java 15. itlt supports all Minecraft versions from the latest all the way back to 1.2.5 and everything in-between.

It seems odd that your modpack's being installed as a normal Vanilla launcher profile rather than Technic's own custom setup. In this case, I recommend turning off useTechnicDisplayName and useTechnicIcon. Could you provide a log of the Technic Launcher itself when launching your pack? It's usually in `~/Application \Support/.technic/logs`

**Comment by martinmaroyan** [comment]
So after I changed the games Java executable to the new openjdk java 1.8 in `/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/bin/java`, the game keeps crashing when I launch it(I'm not 100% sure its because I changed the java executable). Here is latest.log, debug.log, and also crash.log which I made myself it contains what the crash dialogue thing shows when it tells me that Minecraft crashed unexpectedly and it will send the crash log to Apple: 
[latest.log](https://github.com/zlepper/itlt/files/6267377/latest.log)
[debug.log](https://github.com/zlepper/itlt/files/6267379/debug.log)
[crash.txt](https://github.com/zlepper/itlt/files/6267381/crash.txt)

**Comment by PaintNinja** [comment]
Not sure why that's happening, but it's definitely not the itlt mod's fault. 

You could try adding the `-XstartOnFirstThread` JVM arg to the game which might help. If not, downgrade back to 8u74 and live with the icon and title not working.

Alternatively, upgrade to a newer version of Minecraft. Almost all of the mods you're using are either available for 1.16 or not needed. As for the Wynntils mod, you can ask on the Forge or MMD discords for help with upgrading the code. The Forge discord doesn't help people with issues on 1.12, but they're generally happy to help you upgrade as long as you let them know that.

**Comment by martinmaroyan** [comment]
It seems that MacOS doesn't support the custom icons, since Wynntils also said for `Replace vannilla icon` with "MacOS isn't supported"
